ampd/src/mvx/proxy.rs

```diff
@@ -6,11 +6,34 @@ use hex::ToHex;
 use mockall::automock;
 use multiversx_sdk::data::transaction::TransactionOnNetwork;
 use multiversx_sdk::gateway::GatewayProxy;
+use router_api::ChainName;
 
+use crate::monitoring;
+use crate::monitoring::metrics::Msg;
 use crate::types::Hash;
 
 const STATUS_SUCCESS: &str = "success";
 
+pub struct Client {
+    client: GatewayProxy,
```

Comment:
```suggestion
    proxy: GatewayProxy,
```
to stay somewhat consistent with the naming